propolis/src/instance.rs

```diff
@@ -437,6 +478,11 @@ impl Instance {
                     // suspend become stale.
                     inner.suspend_info = None;
                 }
+                State::Migrate(_) => {
+                    let migrate_ctx = inner.migrate_ctx.take().unwrap();
+                    // We explicitly allow the migrate task to run
+                    self.disp.release_one(migrate_ctx);
+                }
```

Comment:
While in the long-term, it'd be nice to separate out the tasks (and/or Ctx objects dispensed to grant instance access) into different classes which dictate when they are run or paused, this seems like a reasonable intermediate step.